flytekit/typing.py

```diff
@@ -78,15 +79,24 @@ def noop():
     ...
 
 
-class FlyteFilePath(os.PathLike):
+class FileFormat(object):
     @classmethod
-    def format(cls) -> str:
-        """
-        When subclassing this, please make sure you do not use the same string as any of the ones already declared
-        in the FlyteFileFormats enum.
-        """
-        return FlyteFileFormats.BASE_FORMAT.value
+    @abstractmethod
+    def extension(cls) -> str:
+        return ""
+
+
+class Text(FileFormat):
```

Comment:
I dont really like this as this implies we have to create a new FileFormat type to work. This I feel is very limit?